### ISSUE TITLE
[analysis] `conj` doesn't provide arglists-strs

### ISSUE THREAD

**Issue by FieryCod** [clj-kondo/clj-kondo#1655]
**version**
clj-kondo v2022.04.08

**platform**
Native

**editor**
Emacs

**problem**
A `conj` function doesn't provide the required information about its signature for `clojure-lsp`, while `conj!` does.

https://clojurians.slack.com/archives/CPABC1H61/p1649790637847509 -- (Issue 1)

Please see @ericdallo response:
> I confirmed the 1. is a kondo issue indeed, clj-kondo returns `:arglist-strs ["[]" "[coll]" "[coll x]"]` for `conj!` but not for `conj`.

**repro**
1. Latest `clojure-lsp` installed.
2. Latest `clj-kondo` installed.
3. Just put `(conj)` in `core.clj` file.
```
(conj! <here_cursor>) shows full signature of a function in a minibuffer, however (conj <here_cursor>) doesn't.
```

**config**
```
;; Emacs config
(use-package! lsp-ui
  :after lsp-mode
  :commands lsp-ui-mode
  :config
  (setq lsp-ui-peek-list-width 60
        lsp-ui-doc-max-width 60
        lsp-ui-doc-enable t
        lsp-ui-peek-fontify 'always
        lsp-ui-sideline-show-code-actions nil))

(use-package! lsp-mode
  :commands lsp
  :hook ((clojure-mode . lsp))
  :config
  (setq lsp-headerline-breadcrumb-enable nil
        lsp-lens-enable t
        lsp-enable-file-watchers t
        lsp-signature-render-documentation t
        lsp-signature-function 'lsp-signature-posframe
        lsp-semantic-tokens-enable t
        lsp-completion-sort-initial-results t ;; check if should keep as t
        lsp-completion-no-cache t
        lsp-completion-use-last-result nil
        lsp-idle-delay 1.0
        lsp-lens-enable t
        lsp-enable-file-watchers t
        lsp-file-watch-threshold 10000
        lsp-signature-auto-activate t
        lsp-enable-indentation nil
        lsp-enable-on-type-formatting nil))
```

**expected behavior**
A signature for `conj` should be provided for `clojure-lsp`.

**Comment by sheluchin** [comment]
I believe this was fixed by #1814 or some other. The definition for `conj` is now returned as:

```clojure
{:fixed-arities #{0 1 2}
 :end-row 91
 :meta {:added "1.0"
        :static true
        :arglists (quote
                   ([]
                    [coll]
                    [coll x]
                    [coll x & xs]))
        :doc "conj[oin]. Returns a new collection with the xs\n    'added'. (conj nil item) returns (item).\n    (conj coll) returns coll. (conj) returns [].\n    The 'addition' may happen at different 'places' depending\n    on the concrete type."}
 :name-end-col 6
 :name-end-row 84
 :name-row 84
 :added "1.0"
 :ns clojure.core
 :name conj
 :defined-by clojure.core/def
 :filename "... /clojure/clojure/src/clj/clojure/core.clj"
 :col 1
 :name-col 2
 :end-col 45
 :arglist-strs ["[]" "[coll]" "[coll x]" "[coll x & xs]"]
 :varargs-min-arity 2
 :doc "conj[oin]. Returns a new collection with the xs\n    'added'. (conj nil item) returns (item).\n    (conj coll) returns coll. (conj) returns [].\n    The 'addition' may happen at different 'places' depending\n    on the concrete type."
 :row 75}
```

**Comment by borkdude** [comment]
@sheluchin Terrific! Thanks
